### PR TITLE
Add Azure DB Server Type (single/flexible)

### LINKED ISSUE #45: Add Azure Flexible Server Type on Mysql and PostgreSQL monitors
Current monitor which allows to supervise the activity of the [Azure Mysql servers](https://github.com/claranet/terraform-datadog-monitors/tree/master/cloud/azure/mysql) and [Azure Postgresl servers](https://github.com/claranet/terraform-datadog-monitors/tree/master/cloud/azure/postgresql), does not allow to monitor "flexible" type servers.

It would be great to get this possibility !

### PR DESCRIPTION
**Pull request by J-Mx**
Closes #45

### CODE REVIEW

File: cloud/azure/mysql/inputs.tf [hunk 1]
```diff
@@ -51,6 +51,12 @@ variable "filter_tags_custom" {
 
 # Azure Databases for MySQL Servers specific variables
 
+variable "server_type" {
```
Comment (BzSpi) [inline cloud/azure/mysql/inputs.tf:54]: Can you add a `validation` block to be sure that input is valid ?

**Comment by Aohzan** [comment]
@J-Mx I think we should add the db type in the monitor name to differentiate the single and the flexible if we need both

**Comment by J-Mx** [comment]
> @J-Mx I think we should add the db type in the monitor name to differentiate the single and the flexible if we need both

Done ✅. The drawback is at doc generation, we find the variable in Readme files.

Regexp [here](https://github.com/claranet/terraform-datadog-scripts/blob/06f6797a48e2e15c9ad4725ade196e4f79a46b96/utils.sh#L49) should be update to avoid that.

**Comment by Aohzan** [comment]
it's already the case for an other monitor https://github.com/claranet/terraform-datadog-monitors/blob/master/database/redis/README.md

**Review by Aohzan — APPROVED** [review]

**Comment by Aohzan** [comment]
thanks @J-Mx

**Pull request merged**